samza-core/src/main/java/org/apache/samza/task/AsyncRunLoop.java

```diff
@@ -374,10 +374,10 @@ public void run() {
         }, commitMs, commitMs, TimeUnit.MILLISECONDS);
       }
 
-      final SystemTimerScheduler timerFactory = task.context().getTimerScheduler();
-      if (timerFactory != null) {
-        timerFactory.registerListener(() -> {
-            state.needTimer();
+      final SystemTimerScheduler timerScheduler = task.context().getTimerScheduler();
```

Comment:
Maybe rename to EpochTimeScheduler/SystemTimeScheduler?

Reply:
There are a couple of refs to "timerScheduler" in TaskContextImpl, but that will be removed in the next PR anyways.